src/main/java/io/neonbee/internal/SharedRegistry.java

```diff
@@ -0,0 +1,118 @@
+package io.neonbee.internal;
+
+import static io.vertx.core.Future.succeededFuture;
+
+import io.neonbee.NeonBee;
+import io.neonbee.logging.LoggingFacade;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.shareddata.AsyncMap;
+import io.vertx.core.shareddata.SharedData;
+
+/**
+ * A registry to manage values in the {@link SharedData} shared map.
+ * <p>
+ * This class is a generic implementation of a registry that can be used to store values in a shared map.
+ *
+ * @param <T> the type of the value to store in the shared registry
+ */
+public class SharedRegistry<T> implements Registry<T> {
+
+    private final LoggingFacade logger = LoggingFacade.create();
+
+    private final SharedData sharedData;
+
+    private final String registryName;
+
+    /**
+     * Create a new {@link WriteSafeRegistry}.
+     *
+     * @param vertx        the {@link Vertx} instance
+     * @param registryName the name of the map registry
+     */
+    public SharedRegistry(Vertx vertx, String registryName) {
+        this(registryName, new SharedDataAccessor(vertx, SharedRegistry.class));
+    }
+
+    /**
+     * Create a new {@link WriteSafeRegistry}.
+     *
+     * @param registryName the name of the map registry
+     * @param sharedData   the shared data
+     */
+    public SharedRegistry(String registryName, SharedData sharedData) {
+        this.registryName = registryName;
+        this.sharedData = sharedData;
+    }
+
+    /**
+     * Register a value in the async shared map of {@link NeonBee} by key.
+     *
+     * @param sharedMapKey the shared map key
+     * @param value        the value to register
+     * @return the future
+     */
+    @Override
+    public Future<Void> register(String sharedMapKey, T value) {
+        logger.info("register value: \"{}\" in shared map: \"{}\"", sharedMapKey, value);
```

Comment:
I'm not a big fan of this `if (logger.isInfoEnabled())` approach, (but this just my own opinion). However, as I can see you're following it, to me it would make more sense to be consistent and have it across the rest of your changes as well. (Both here and below).

Reply:
Normally PMD would complain here. But as it does not recognise our logger, PMD ‘overlooks’ this. In general, however, I'm not a fan of this either and would rather have a logger that evaluates lambda expressions as parameters.